src/java/main/org/apache/zookeeper/Version.java

```diff
@@ -20,7 +20,7 @@
 
 public class Version implements org.apache.zookeeper.version.Info {
 
-    public static int getRevision() {
+    public static String getRevision() {
```

Comment:
I have checked with my HBase colleagues, they said no dependency with "revision". From ZK point of view, not required to maintain backward compatibility as this is not exposed interface. Still, if any concerns, then we can deprecate and create a new GIT_REVISION or REVISION_HASH etc as this change will go to stable branch-3.4